src/schedule.rs

```diff
@@ -87,23 +87,28 @@ impl Schedule {
 
                             for second in self.fields.seconds.ordinals().range(second_range).cloned() {
                                 let timezone = after.timezone();
-                                let candidate = if let Some(candidate) = timezone
-                                    .ymd(year as i32, month, day_of_month)
-                                    .and_hms_opt(hour, minute, second)
+                                println!("{year} {month} {day_of_month} {hour} {minute} {second}");
+                                let candidate = if let Some(candidate) =
+                                    NaiveDate::from_ymd_opt(year as i32, month, day_of_month)
+                                        .unwrap()
+                                        .and_hms_opt(hour, minute, second)
                                 {
-                                    candidate
+                                    match candidate.and_local_timezone(timezone) {
+                                        LocalResult::None => continue,
+                                        some => some,
+                                    }
                                 } else {
                                     continue;
                                 };
                                 if !self
                                     .fields
                                     .days_of_week
                                     .ordinals()
-                                    .contains(&candidate.weekday().number_from_sunday())
+                                    .contains(&candidate.clone().latest().unwrap().weekday().number_from_sunday())
```

Comment:
Being a `LocalResult`, this needs to be reduced to a single time; Since we know it's not `None`, we know `.unwrap()` won't panic.